index.html

```diff
@@ -451,11 +451,13 @@ <h2 id="collections">Collection Functions (Arrays or Objects)</h2>
         <b class="header">each</b><code>_.each(list, iteratee, [context])</code>
         <span class="alias">Alias: <b>forEach</b></span>
         <br />
-        Iterates over a <b>list</b> of elements, yielding each in turn to an <b>iteratee</b>
-        function. The <b>iteratee</b> is bound to the <b>context</b> object, if one is
+        Iterates over a <b>list</b> of elements, yielding each in turn to an
+        <a href="#iteratee"><b>iteratee</b></a> function.
```

Comment:
After a double-check on #1864, I've seen a comment from @megawac stating 

> Each and reduce don't use iteratee - they use createCallback

So this is definitively incorrect. Do we just remove the link or do we adjust the wording here?